docs/dev/data_validation.rst

```diff
@@ -0,0 +1,688 @@
+
+--------------------------------------------------------------------------------
+Running the data validation tests
+--------------------------------------------------------------------------------
+
+Setting up dbt
+~~~~~~~~~~~~~~
+
+The dbt directory contains the PUDL dbt project which manages our `data tests
+<https://docs.getdbt.com/docs/build/data-tests>`_. To run dbt you'll need to have the
+``pudl-dev`` conda environment activated (see :doc:`dev_setup`). Before running the
+tests you'll want to run the following from within the ``dbt/`` directory in your pudl
+repo:
+
+.. code-block:: bash
+
+    # Install additional dbt-specific dependencies in the dbt project
+    dbt deps
+    # Materializes dbt-specific "seed" tables from CSVs in the seeds/ directory
+    # In our case, these contain the expected row counts for PUDL tables
+    dbt seed
+
+The data validation tests run on the Parquet outputs that are in your
+``$PUDL_OUTPUT/parquet/`` directory. It's important that you ensure the outputs you're
+testing are actually the result of the code on your current branch, otherwise you may
+be surprised when the data test fails in CI or the nightly builds.
+
+.. tip::
+
+   If you want to run the tests against the fast ETL outputs that are generated by the
+   integration tests without wiping out your local full ETL outputs, you can temporarily
+   set the ``$PUDL_OUTPUT`` environment variable to point at the tmpdir created by
+   ``pytest``. This is printed out at the beginning of the ``pytest`` run.
+
+Running dbt directly
+~~~~~~~~~~~~~~~~~~~~
+
+dbt has its own much more `extensive documentation <https://docs.getdbt.com/>`_. PUDL
+uses only a small subset of its features, and our use case is slightly unusual, since we
+rely on Dagster to coordinate our data pipeline, and are only using dbt for data
+validation. Some quirks of our setup to be aware of:
+
+* From dbt's point of view, the PUDL tables are
+  `sources <https://docs.getdbt.com/docs/build/sources>`_ -- external tables about which
+  it knows very little other than the table and column names. It assumes the tables will
+  will be available, rather than trying to create them. In a typical dbt project, most
+  tables would be defined as `models <https://docs.getdbt.com/docs/build/models>`_ which
+  are somewhat analogous to `Dagster assets
+  <https://docs.dagster.io/guides/build/assets/defining-assets>`_.
+* As a SQL-based tool, dbt generally expects to be querying a database. However, in our
+  case the tables are stored as Apache Parquet files, which we query with SQL via
+  DuckDB. This means some of dbt's functionality is not available. For example, we can't
+  use `the dbt adapter object
+  <https://docs.getdbt.com/reference/dbt-jinja-functions/adapter>`_ in our test
+  definitions because it relies on being able to access the underlying database schema,
+  which doesn't exist (at least, not within the dbt context).
```

Comment:
Are there other quirks that folks have run into that they'd like to add to this list? Is this the right place for it?